client/finality-grandpa/src/finality_proof.rs

```diff
@@ -148,8 +151,82 @@ impl<Block: BlockT> AuthoritySetForFinalityChecker<Block> for Arc<dyn FetchCheck
 	}
 }
 
+/// Justification for a finalized block.
+#[derive(Clone, Serialize, Deserialize)]
+pub struct JustificationNotification<Block: BlockT> {
```

Comment:
IMO, this type should be defined in the rpc module. We can just use `(Block::Header, Justification)` in this crate. Also not sure if we need the header here or if just the block hash is enough. The fianlized block number and block hash is already defined in the justification (although for convenience I guess it might be useful to have it on a top-level like this).